com.civicrm.hrjobroles/js/src/job-roles/controllers/job-roles.controller.js

```diff
@@ -524,6 +538,8 @@ define([
                       value: data.values[i]['value'],
                       is_active: data.values[i]['is_active']
                     };
+                    // Store the Department types so we can reuse later
```

Comment:
We are always encouraged to refactor old code (which is a duplicate feast at the moment). I would suggest smth like that, but with caution. Please make sure there are test suites for that part of code, if not, maybe this change is not worth it.

```js
// Replacement starting from `var DepartmentList = {};`
var optionsData = {
  'hrjc_department': { storage: 'DepartmentsData', data: {} },
  ...,
  'cost_centres': { storage: 'CostCenters', data: [] }
};

angular.forEach(data['optionGroupData'], function (optionGroupId, optionGroupName) {
  for (var i = 0; i < data.count; i++) {
    if (optionGroupId === data.values[i]['option_group_id']) {
      if (optionGroupName === 'cost_centres') {
        optionsData[optionGroupName].push({
          id: data.values[i]['id'],
          title: data.values[i]['label'],
          weight: data.values[i]['weight'],
          is_active: data.values[i]['is_active']
        });
      } else {
        optionsData[optionGroupName].data[data.values[i]['value']] = {
          id: data.values[i]['id'],
          title: data.values[i]['label'],
          value: data.values[i]['value'],
          is_active: data.values[i]['is_active']
        };
      }

      vm[optionsData[optionGroupName].storage] = getActiveValues(optionsData[optionGroupName].data);
    }
  }
});
```

cc @AkA84

Reply:
@ajesamson actually, since we are lacking suites that test `vm. DepartmentsData` etc, I think this function should be refactored after tests are added. Writing this because in the comment above I was not sure if it worth it, now I realise it is.

cc @AkA84